File: crates/xmtp_sdk/conformance/kotlin/src/main/kotlin/ConsumerPositive.kt
import uniffi.xmtp_sdk.*

fun consumePositive(
    id: ConversationID,
    conversation: Conversation,
    content: MessageContent,
): ConversationID {
    val narrowed: ConversationID =
        when (conversation) {
            is Conversation.Group -> conversation.group.id()
            is Conversation.Dm -> conversation.dm.id()
        }
    if (content is MessageContent.Custom) {
        val encoded: EncodedContent = content.encoded
        check(encoded.type.typeID.isNotEmpty())
    }
    return if (narrowed == id) id else narrowed
}

fun consumeStandardIDs(content: StandardContent): MessageID? =
    when (content) {
        is StandardContent.Reaction -> {
            val inbox: InboxID? = content.referenceInboxID
            check(inbox == null || inbox.toString().isNotEmpty())
            content.reference
        }

        is StandardContent.Reply -> {
            content.reference
        }

        is StandardContent.DeleteMessage -> {
            content.messageID
        }

        else -> {
            null
        }
    }

File: crates/xmtp_sdk/conformance/swift/Sources/Conformance/ConsumerPositive.swift
import XmtpSdk

func consumePositive(_ id: ConversationID, _ conversation: Conversation, _ content: MessageContent) -> ConversationID {
    let narrowed: ConversationID
    switch conversation {
    case let .group(group): narrowed = group.id()
    case let .dm(dm): narrowed = dm.id()
    }
    if case let .custom(encoded) = content {
        let _: EncodedContent = encoded
    }
    return narrowed == id ? id : narrowed
}

func consumeStandardIDs(_ content: StandardContent) -> MessageID? {
    switch content {
    case let .reaction(reference, inboxID, _):
        let _: InboxID? = inboxID
        return reference
    case let .reply(reference, inboxID, _):
        let _: InboxID? = inboxID
        return reference
    case let .deleteMessage(messageID):
        let id: MessageID = messageID
        return id
    default:
        return nil
    }
}
